iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/NodeBreakerTopology.java

```diff
@@ -38,26 +35,14 @@ private NodeBreakerTopology() {
     public static void removeIsolatedSwitches(VoltageLevel.NodeBreakerView topo) {
         Objects.requireNonNull(topo, "Node breaker topology view is null.");
 
-        int[] nodes = topo.getNodes();
-        final TIntSet encountered = new TIntHashSet();
         final Set<Switch> encounteredSwitches = new HashSet<>();
+        int[] nodesWithTerminal = Arrays.stream(topo.getNodes()).filter(n -> topo.getTerminal(n) != null).toArray();
+        topo.traverse(nodesWithTerminal, (n1, sw, n2) -> {
+            encounteredSwitches.add(sw); // the traversing started from a terminal, thus the switch is not isolated
+            return TraverseResult.CONTINUE; // if n2 has a terminal, we could also choose to stop as it will be or has already been traversed
```

Comment:
Shouldn't we stop when we encounter another terminal? If we don't, it seems to me that if there is a loop, the traverser never stops (but maybe I am mistaken)